File: Examples/Java/SDK/src/main/java/com/aspose/cells/cloud/examples/chart/AddChart.java
package com.aspose.cells.cloud.examples.chart;

import java.io.IOException;
import java.nio.file.Path;

import com.aspose.cells.api.CellsApi;
import com.aspose.cells.cloud.examples.Configuration;
import com.aspose.cells.cloud.examples.Utils;
import com.aspose.cells.model.ChartsResponse;
import com.aspose.storage.api.StorageApi;

public class AddChart {

    public static void main(String... args) throws IOException {
        // For complete examples and data files, please go to
        // https://github.com/aspose-cells/Aspose.Cells-for-Cloud
        // ExStart: 1
        
        String fileName = "Sample_Test_Book.xls";
        String sheetName = "Sheet6";
        String chartType = "bar";
        int upperLeftRow = 12;
        int upperLeftColumn = 12;
        int lowerRightRow = 20;
        int lowerRightColumn = 20;
        String area = "A1:A3";
        Boolean isVertical = false;
        String categoryData = "";
        Boolean isAutoGetSerialName = true;
        String title = "SalesState";
        String storage = "";
        String folder = "";
        Path inputFile = Utils.getPath(AddChart.class, "Sample.xlsx");

        try {
            StorageApi storageApi = new StorageApi(Configuration.apiKey, Configuration.appSID, true);

            // Instantiate Aspose Words API SDK
            CellsApi cellsApi = new CellsApi(Configuration.apiKey, Configuration.appSID, true);
            // Upload source file to aspose cloud storage
            storageApi.PutCreate(fileName, "", "", inputFile.toFile());

            // Invoke Aspose.Cells Cloud SDK API to add chart in worksheet
            ChartsResponse apiResponse = cellsApi.PutWorksheetAddChart(fileName, sheetName, chartType, upperLeftRow,
                    upperLeftColumn, lowerRightRow, lowerRightColumn, area, isVertical, categoryData,
                    isAutoGetSerialName, title, storage, folder);

            if (apiResponse != null && apiResponse.getStatus().equals("OK")) {
                System.out.println("Add a Chart in a Worksheet, Done!");

            }
        } catch (Exception ex) {
            ex.printStackTrace();
        }
        // ExEnd: 1

    }
}

File: Examples/DotNet/CSharp/Charts/AddChart.cs
using System;
using Com.Aspose.Cells.Api;
using Com.Aspose.Cells.Model;
using Com.Aspose.Storage.Api;

namespace Charts
{
    class AddChart
    {
        public static void Run()
        {
            // ExStart:1
            CellsApi cellsApi = new CellsApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);
            StorageApi storageApi = new StorageApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);

            String fileName = "Sample_Test_Book.xls";
            String sheetName = "Sheet6";
            String chartType = "bar";
            int upperLeftRow = 12;
            int upperLeftColumn = 12;
            int lowerRightRow = 20;
            int lowerRightColumn = 20;
            String area = "A1:A3";
            Boolean isVertical = false;
            String categoryData = "";
            Boolean isAutoGetSerialName = true;
            String title = "SalesState";
            String storage = "";
            String folder = "";

            try
            {
                // Upload source file to aspose cloud storage
                storageApi.PutCreate(fileName, "", "", System.IO.File.ReadAllBytes(Common.GetDataDir() + fileName));

                // Invoke Aspose.Cells Cloud SDK API to add chart in worksheet
                ChartsResponse apiResponse = cellsApi.PutWorksheetAddChart(fileName, sheetName, chartType, upperLeftRow, upperLeftColumn, lowerRightRow, lowerRightColumn, area, isVertical, categoryData, isAutoGetSerialName, title, storage, folder);

                if (apiResponse != null && apiResponse.Status.Equals("OK"))
                {
                    Console.WriteLine("Add a Chart in a Worksheet, Done!");
                    Console.ReadKey();
                }
            }
            catch (Exception ex)
            {
                System.Diagnostics.Debug.WriteLine("error:" + ex.Message + "\n" + ex.StackTrace);
            }
            // ExEnd:1
        }
    }
}
